actionpack/lib/action_controller/metal/conditional_get.rb

```diff
@@ -77,18 +77,16 @@ def etag(&etagger)
     #
     #   before_action { fresh_when @article, template: 'widgets/show' }
     #
-    def fresh_when(record_or_options, additional_options = {})
-      if record_or_options.is_a? Hash
-        options = record_or_options
-        options.assert_valid_keys(:etag, :last_modified, :public, :template)
-      else
-        record  = record_or_options
-        options = { etag: record, last_modified: record.try(:updated_at) }.merge!(additional_options)
+    def fresh_when(record = nil, etag: record, last_modified: nil, public: false, template: nil)
+      last_modified ||= record.updated_at if record
```

Comment:
However, there are currently no tests that invoke `fresh_when` or `stale?` with an object that does not define `updated_at`, and even the documentation says:

>  You can also just pass a record where `last_modified` will be set by calling `updated_at` and the `etag` by passing the object itself

So removing the `try` might be acceptable according to both tests and docs.
If you decide to do so, I suggest you remove it in a separate commit, because everything else
in your commit is 100% backward-compatible